File: python/selection_sort.py
num=[5,3,7,1,8,9,2]

for i in range(len(num)):
    min=num[i]
    index=i
    j=i+1
    while(j<len(num)):
        if(num[j]<min):
            min=num[j]
            index=j
        j+=1
    if(index!=i):
        temp=num[i]
        num[i]=num[index]
        num[index]=temp

print(num)

File: java/selection_sort.java
public class selection_sort{
    public static void main(String[] args) {
        int[] num={7,3,8,1,9,4};
        int min=num[0];
        for(int i=0;i<num.length;i++){
            int index=i;
            min=num[i];
            for(int j=i+1;j<num.length;j++){
                if(num[j]<min){
                    index=j;
                    min=num[j];
                }
            }
            if(index!=i){
                int temp=num[i];
                num[i]=num[index];
                num[index]=temp;
            }
        }
        for(int i:num){
            System.out.print(i);
        }
    }
}